code/modules/client/client procs.dm

```diff
@@ -333,6 +333,8 @@
 	if(!prefs)
 		prefs = new /datum/preferences(src)
 		preferences_datums[ckey] = prefs
+	else
+		prefs.parent = src
```

Comment:
What's the need for this?

Reply:
When you log out, your client is gone which means the preferences loses its client. Now, when you log back in the parent client will be set properly. This has been around for forever, but I guess this is the first time anything needs to call the client.